gobblin-modules/gobblin-kafka-common/src/main/java/org/apache/gobblin/kafka/schemareg/HttpClientFactory.java

```diff
@@ -30,9 +32,27 @@
  */
 public class HttpClientFactory extends BasePooledObjectFactory<HttpClient>{
 
+  private Optional<Integer> soTimeout = Optional.empty();
```

Comment:
I think it is cleaner to use int here and allow -1 to be used to indicate that the default be used. Or if you want to use optional then change `HttpClientFactory(int socketTimeout, int connectionTimeout)` to use optional to allow flexibility in setting one of the timeouts without setting both.